high_card.rb

```diff
@@ -2,4 +2,67 @@
 ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
+cards = []
+
+ranks.each do |rank|
+  suits.each do |suit|
+    cards.push([rank,suit])
+  end
+end
+
+cards = cards.shuffle
+
 players = []
+players_cards = []
+
+puts 'How many players are playing today?'
+num_players = gets.chomp.to_i
+
+# question - is it best practice to always pass arugments to fuctions/methods
+# rather than using glabal variables (both JS and ruby)?
```

Comment:
Yep. Programmers really hate on global variables, and with good reason -- they can really clog up your memory because they don't get garbage collected until the very end of the script.